cli/src/main/java/com/liferay/blade/cli/BladeSettings.java

```diff
@@ -31,102 +42,282 @@
  */
 public class BladeSettings {
 
-	public BladeSettings(File settingsFile) throws IOException {
-		_settingsFile = settingsFile;
+	public static String[] getUsageArgs(String[] args, Class<? extends BaseArgs> clazz, Collection<ParameterDescription> parameters) {
+		Class<?> localClazz = clazz;
+		List<String> argsList = new ArrayList<>(Arrays.asList(args));
+		List<String> newArgsList = new ArrayList<>();
+		String finalParameter = null;
+		while (BaseArgs.class.isAssignableFrom(localClazz)) {
+			for (Field field : localClazz.getDeclaredFields()) {
+				Parameter annotation = field.getAnnotation(Parameter.class);
+				
+
+				if (annotation != null) {
+					String[] names = annotation.names();
+
+					if ((names != null) && (names.length > 0)) {
+						
+						String nameUsed = null;
+
+						for (String name : names) {
+							if (argsList.contains(name)) {
+								nameUsed = name;
+
+								break;
+							}
+						}
+
+						if (nameUsed != null) {
+							CollectUsage collectUsageAnnotation = field.getAnnotation(CollectUsage.class);
+							
+
+							if (collectUsageAnnotation != null) {
+								Class<?> type = field.getType();
+
+								if (!type.equals(boolean.class)) {
+									newArgsList.add(nameUsed);
+									
+
+									if (collectUsageAnnotation.censor()) {
+										newArgsList.add("<censored>");
+									}
+									else {
+										newArgsList.add(argsList.get(argsList.indexOf(nameUsed) + 1));
+									}
+								}
+								else {
+									newArgsList.add(nameUsed);
+								}
+							}
+						}
+					}
+					else {
+						CollectUsage collectUsageAnnotation = field.getAnnotation(CollectUsage.class);
+						
+
+						if (collectUsageAnnotation != null) {
+								
+
+							if (collectUsageAnnotation.censor()) {
+								finalParameter = "<final parameter censored>";
+							}
+							else {
+								finalParameter = argsList.get(newArgsList.size() - 1);
+							}
+						}
+					}
+				}
+			}
+
+			localClazz = localClazz.getSuperclass();
+		}
+
+		if (finalParameter != null) {
+			newArgsList.add(finalParameter);
+		}
 
-		if (_settingsFile.exists()) {
+		return newArgsList.toArray(new String[0]);
+	}
+
+	public BladeSettings(BladeCLI bladeCLI, File bladeUserHomeDir, File... settingsFiles) throws IOException {
```

Comment:
Why does this have multiple settings files?  This seems overly complicated.  Do we really need multiple settings files support?